src/Cone.cpp

```diff
@@ -6,178 +6,187 @@
-	} else {
-		return LineIntersection(start, side2, lineStart, lineEnd - lineStart);
-	}
-	
-}
-
-bool Cone::DoConesOverlap(Cone otherCone) const {
-	if (this->IsWithinCone(otherCone.side1 + otherCone.start) || this->IsWithinCone(otherCone.side2 + otherCone.start) || otherCone.IsWithinCone(side1 + start) || otherCone.IsWithinCone(side2 + start)) {
-		return true;
-	} else {
-		return false;
-	}
-}
-
-Cone Cone::MergeCones(Cone otherCone) const {
-	if (!DoConesOverlap(otherCone)) {
-		std::cout << "no overlap" << std::endl;
-		return *this;
-	}
-
-	if (this->IsWithinCone(otherCone.side1 + otherCone.start) && this->IsWithinCone(otherCone.side2 + otherCone.start)) {
-		return *this;
-	}
-	if (otherCone.IsWithinCone(side1 + start) && otherCone.IsWithinCone(side2 + start)) {
-		return otherCone;
-
-	}
-
-	double angleDiff1 = fabs(cleanAngle(side1.angle() - otherCone.side2.angle()));
-	double angleDiff2 = fabs(cleanAngle(side2.angle() - otherCone.side1.angle()));
-
-	if (angleDiff1 > angleDiff2) {
-		return Cone(start, side1, otherCone.side2);
-	} else {
-		return Cone(start, otherCone.side1, side2);
-	}
-}
-
-Cone::~Cone(){}
+    Cone::Cone(Vector2 startPoint, Vector2 centerPoint, double distance) {
+        start = startPoint;
+        center = centerPoint;
+        radius = distance;
+        if (radius > 0) {
+            angle = 2 * atan((radius) / (center - start).length());
+        } else {
+            // ROS_WARN("radius for cone invalid!");
+        }
+        side1 = (center - start).rotate(0.5 * angle);
+        side2 = (center - start).rotate(-0.5 * angle);
+    }
+
+    Cone::Cone(Vector2 startPoint, Vector2 sideA, Vector2 sideB) {
+        start = startPoint;
+        if (sideA.length() > sideB.length()) {
+            side1 = sideA;
+            side2 = sideB.stretchToLength(sideA.length());
+        } else {
+            side1 = sideA.stretchToLength(sideB.length());
+            side2 = sideB;
+        }
+        angle = cleanAngle(side1.angle() - side2.angle());
+        center = side2.rotate(0.5 * angle) + start;
+        angle = fabs(angle);
+        radius = (start + side1 - center).length();
+    }
+
+    bool Cone::IsWithinCone(Vector2 point) const noexcept {
+        return IsWithinCone(point, 0.0);
+    }
+
+    bool Cone::IsWithinCone(Vector2 point, double marginRadius) const noexcept {
+        Vector2 vectorToPoint = point - start;
+        Vector2 vectorToCenter = center - start;
+        double extraAngle = atan(marginRadius / vectorToPoint.length());
+
+        if (fabs(rtt::cleanAngle(vectorToPoint.angle() - vectorToCenter.angle())) < (0.5 * angle + extraAngle)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+
+    bool Cone::IsWithinField(Vector2 point) const noexcept {
+        /**
+         * Guys, no return value is undefined behavior and can crash your program
+         * Please pay attention to this stuff
+         */
```

Comment:
Yes. Pretty sure this class is never even constructed anywhere, though.

Reply:
Anyways, comments like this are fine but should be on github and not in the code  ^^